src/components/Auth/ProductForm.js

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react';
+import Form from '../Shared/Form/Form';
+import productActions from '../../store/product/action';
+import { useDispatch } from 'react-redux';
+
+function ProductForm() {
+  const [isSubmitted, setIsSubmitted] = useState('');
+  const [allInputsObject, setAllInputsObject] = useState({});
+  const dispatch = useDispatch();
+
+  const onSubmit = (event, object) => {
+    event.preventDefault();
+    setIsSubmitted(true);
+
+    if (
+      object.title !== '' &&
+      object.price !== isNaN() &&
+      object.category !== '' &&
+      object.imageUrl !== ''
+    ) {
+      dispatch(productActions.addProduct(object));
+      setAllInputsObject({ ...object });
+    } else {
+      event.preventDefault();
+    }
+  };
+
+  return (
+    <div className="d-flex">
+      <Form
+        type="text"
+        title="Add New Product"
+        inputArray={[
+          {
+            title: 'Title',
+            name: 'title',
+            type: 'input',
+            placeholder: '',
+          },
+          {
+            title: 'Price',
+            name: 'price',
+            type: 'price',
+            placeholder: '',
+          },
+          {
+            title: 'Category',
+            name: 'category',
+            type: 'dropdown',
+            dropdownOptions: [
+              { name: 'Bread', id: 'bread' },
+              { name: 'Vegetables', id: 'vegetables' },
+              { name: 'Fruits', id: 'fruits' },
+              { name: 'Dairy', id: 'dairy' },
+            ],
+            placeholder: '',
+          },
+          {
+            title: 'Image Url',
+            name: 'imageUrl',
+            type: 'input',
+            placeholder: '',
+          },
+        ]}
+        primaryButton={{ title: 'Save Product' }}
+        restart={() => {
```

Comment:
can you make this an optional parameter so you don't have to pass in a function that doesnt do anything